PWGHF/D2H/Tasks/taskCharmPolarisation.cxx

```diff
@@ -1576,7 +1576,7 @@ struct HfTaskCharmPolarisation {
       // variable definition
       float pxDau{-1000.f}, pyDau{-1000.f}, pzDau{-1000.f};
       float pxCharmHad{-1000.f}, pyCharmHad{-1000.f}, pzCharmHad{-1000.f};
-      float massDau{0.f}, invMassCharmHad{0.f}, invMassCharmHadForSparse{0.f}, invMassD0{0.f}, invMassKPiLc{0.f}, invMassPKLc{0.f}, invMassPPiLc{0.f};
+      double massDau{0.}, invMassCharmHad{0.}, invMassCharmHadForSparse{0.}, invMassD0{0.}, invMassKPiLc{0.}, invMassPKLc{0.}, invMassPPiLc{0.};
```

Comment:
Regarding this, it is not clear to me what is the benefit of having the mass as `double` when we compute it starting from `float` values (momenta and pdg masses).
Wouldn't it be better to have the function in `RecoDecay` returning the same type as the inputs instead of always `double`?
https://github.com/AliceO2Group/O2Physics/blob/689968149d4069e09f982edb23834b4e45d9a67b/Common/Core/RecoDecay.h#L449